src/Provider/Chain/Chain.php

```diff
@@ -35,9 +45,21 @@ final class Chain implements Provider, LoggerAwareInterface
     /**
      * @param Provider[] $providers
      */
-    public function __construct(array $providers = [])
+    public function __construct(array $providers = [], string $geocodeQueryLogLevel = 'alert', string $reverseQueryLogLevel = 'alert')
```

Comment:
Do we need a different log level on both methods, or would be it be adequate to set it for the class?
Since we can't pass the logger to the constructor, i'd skip passing the log levels there.

Reply:
>Do we need a different log level on both methods, or would be it be adequate to set it for the class?

I would "be happy" with the ability to differentiate between the two types of functionality:

- geocode
- reverse geocode

Other users also might have other use cases / needs. 

>Since we can't pass the logger to the constructor, i'd skip passing the log levels there.

Not sure how this relates with each other? I do agree we should use the `LogLevels::ALERT` constant as default. 

But feel

Reply:
>I would "be happy" with the ability to differentiate between the two types of functionality:

On second thought, I guess there's no reason to reduce configurability.

> Not sure how this relates with each other? I do agree we should use the LogLevel::ALERT constant as default.

I was meaning, keeping it consistent by having the loggers set by setters, rather than passed in the constructor.

Keep in mind i'm not the owner of this repo, so take my ideas with a grain of salt :smile:

Reply:
>I was meaning, keeping it consistent by having the loggers set by setters, rather than passed in the constructor.

Aah, ok, understood. Yeah, might be better to not pollute the constructor too much.